### PR TITLE
Support ``visit_Module`` in ``pass_array_by_data.cpp``

### PR DESCRIPTION
**Pull request by czgdp1807**
https://github.com/lfortran/lfortran/pull/1306

cc: @Shaikh-Ubaid

### CODE REVIEW

**Comment by czgdp1807** [comment]
I think I have figured out why `minpack` compilation fails. Will make a fix, add a test for it as well.

File: integration_tests/minpack_02.f90 [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+implicit none
+
+integer, parameter :: dp=kind(0d0)
+
+contains
+
+subroutine check_deriv(x, fvec, fjac)
+real(dp), intent(in) :: x(:)
+real(dp), intent(out) :: fvec(:), fjac(:,:)
+real(dp) :: xp(size(x)), fvecp(size(fvec)), err(size(fvec))
```
Comment (czgdp1807) [inline integration_tests/minpack_02.f90:11]: Should we use `FunctionParam` for this?

Comment (certik) [reply integration_tests/minpack_02.f90:11]: Yes, for sure. But after we finish the FunctionParam PR.

**Review by certik — APPROVED** [review]
I think this is fine, I haven't reviewed every detail, but haven't seen any major issue.

**Comment by czgdp1807** [comment]
Sure. I will merge it tomorrow after cleaning up the git history. Until then if you find any issue or any explanation/documentation is needed let me know.

**Comment by certik** [comment]
@Shaikh-Ubaid and @Pranavchiku can you please also review it?

**Review by Pranavchiku — APPROVED** [review]
Looks good to me!

**Comment by Shaikh-Ubaid** [comment]
**doubt:** In https://github.com/lfortran/lfortran/pull/1306#discussion_r1125511140 we were removing the old external symbols and adding new ones. Similarly, in this PR do we need not remove the old external symbols? (From the PR changes it seems we are adding new external symbols. I am unsure if we removed the old external symbols.)

**Review by Shaikh-Ubaid — APPROVED** [review]
I shared a doubt. The PR looks good to me. Thank you so much for this!

**Comment by Pranavchiku** [comment]
> do we need not remove the old external symbols?

As far as I know, no expression or statements will be referring to those old external symbols ( of functions ), and later on, pass `remove_unused_functions` (not sure of name) removes unused functions from ASR hence it makes no trouble to us.

**Comment by czgdp1807** [comment]
> we were removing the old external symbols and adding new ones.

Actually I think in ASR verify we should make sure that any `ExternalSymbol::m_external` should actually exist in some scope. If not then the `ExternalSymbol` itself is dead and we should raise an error. Let me try adding it. In this PR this kind of check should fail, if it doesn't then we are safe to merge.

**Pull request merged**